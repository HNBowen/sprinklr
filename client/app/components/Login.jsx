import React from 'react'
import PropTypes from 'prop-types'

const Login = (props) => {
  return (
      <div>
        <form>
          username: <input type="text" name="username"></input>
          password: <input type="text" name="password"></input>
          <button onClick={props.handleLogin}>Login</button>
        </form>
        <button onClick={props.handleRegister}>Register</button>
      </div>
    )
}

Login.propTypes = {
  handleLogin: PropTypes.func.isRequired,
  handleRegister: PropTypes.func.isRequired
}

export default Login