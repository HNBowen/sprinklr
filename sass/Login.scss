/* variables for font styles*/

$font-stack: "American Typewriter", sans-serif;

@mixin font-color($color) {
  -webkit-text-fill-color: $color;
  -moz-text-fill-color: $color;
  -ms-text-fill-color: $color;
  text-fill-color: $color;
}

/*/// Stroke font-character
/// @param  {Integer} $stroke - Stroke width
/// @param  {Color}   $color  - Stroke color
/// @return {List}            - text-shadow list*/

@function stroke($stroke, $color) {
  $shadow: ();
  $from: $stroke*-1;
  @for $i from $from through $stroke {
   @for $j from $from through $stroke {
      $shadow: append($shadow, $i*1px $j*1px 0 $color, comma);
    }
  }
  @return $shadow;
}
/*/// Stroke font-character
/// @param  {Integer} $stroke - Stroke width
/// @param  {Color}   $color  - Stroke color
/// @return {Style}           - text-shadow*/
@mixin stroke($stroke, $color) {
  text-shadow: stroke($stroke, $color);
}


/*// Flexbox display*/
@mixin flexbox() {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
}

/*// Flex Flow Direction
// - applies to: flex containers
// row | row-reverse | column | column-reverse*/
@mixin flex-direction($direction) {
  -webkit-flex-direction: $direction;
     -moz-flex-direction: $direction;
      -ms-flex-direction: $direction;
          flex-direction: $direction;
}


/*// Axis Alignment
// - applies to: flex containers
// flex-start | flex-end | center | space-between | space-around */
@mixin justify-content($justify) {
  -webkit-justify-content: $justify;
     -moz-justify-content: $justify;
      -ms-justify-content: $justify;
          justify-content: $justify;
            -ms-flex-pack: $justify;
}

/*// Cross-axis Alignment
// - applies to: flex containers
// flex-start | flex-end | center | baseline | stretch */
@mixin align-items($align) {
  -webkit-align-items: $align;
     -moz-align-items: $align;
      -ms-align-items: $align;
          align-items: $align;
}

/* mixin for border radius */

@mixin border-radius($radius) {
  -webkit-border-radius: $radius;
     -moz-border-radius: $radius;
      -ms-border-radius: $radius;
          border-radius: $radius;
}


.login-page {
  background-image: url("../assets/brina-blum-215893.jpg");
  background-size: cover;
  background-position: center;
  width: 100vw;
  height: 100vh;
  margin: 0;
  font-family: $font-stack;
  @include font-color(white);
  @include stroke(1, black)
  @include flexbox();
  @include flex-direction(column);
  @include justify-content(center);
  @include align-items(center);
}

.login-title {
  @include border-radius(7px)
  background-color: rgba(249, 218, 208, 0.5);
}

.login-form {
  @include flexbox();
  @include flex-direction(column);
  @include justify-content(center);
  @include align-items(center)
  @include border-radius(7px)
  background-color: rgba(249, 218, 208, 0.5);
}

.login-button {
  font-family: $font-stack;
  @include stroke(1, black)
}

.register-link {
  @include font-color(white);
}

/*
  
  - .login-page background image should be centered
  - .login-form should be centered
  - .login-title should be above the .login-form
  - same font family for all text -> save in a var
  - same font fill and border -> save in a var
  - .regsiter-link should be below .login-form

*/