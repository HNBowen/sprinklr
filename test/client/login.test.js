import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import Login from '../../client/app/components/Login.jsx'

describe('Login', function() {

  it('should render correctly', function() {
    let wrapper = shallow(<Login handleLogin={()=>{}} handleRegister={()=>{}}/>)
    jestExpect(shallowToJson(wrapper)).toMatchSnapshot();
  })

  it('should render with an input field for username', function() {
    let wrapper = shallow(<Login handleLogin={()=>{}} handleRegister={()=>{}}/>)

    let usernameFieldWrapper = wrapper.find('input').get(0)

    expect(usernameFieldWrapper).to.exist;
    expect(usernameFieldWrapper.props.name).to.equal('username')
  })

  it('should render with an input field for password', function() {
    let wrapper = shallow(<Login handleLogin={()=>{}} handleRegister={()=>{}}/>)

    let passwordFieldWrapper = wrapper.find('input').get(1);

    expect(passwordFieldWrapper).to.exist;
    expect(passwordFieldWrapper.props.name).to.equal('password')
  })

  it('should render with Login and Register buttons', function() {

    let wrapper = shallow(<Login handleLogin={()=>{}} handleRegister={()=>{}}/>)

    let loginButtonWrapper = wrapper.find('button').get(0);
    let registerButtonWrapper = wrapper.find('button').get(1);
    
    expect(loginButtonWrapper.props.children).to.equal('Login')
    expect(registerButtonWrapper.props.children).to.equal('Register')

  })

  it('should call click handlers when buttons are clicked', function() {
    let loginMock = jest.fn();
    let registerMock = jest.fn();

    let wrapper = shallow(<Login handleLogin={loginMock} handleRegister={registerMock} />)

    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')

    expect(loginMock.mock.calls.length).to.equal(1);
    expect(registerMock.mock.calls.length).to.equal(1);
  })

})
