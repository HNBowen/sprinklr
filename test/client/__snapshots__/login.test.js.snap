// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div>
  <form>
    username: 
    <input
      name="username"
      type="text"
    />
    password: 
    <input
      name="password"
      type="text"
    />
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </form>
  <button
    onClick={[Function]}
  >
    Register
  </button>
</div>
`;
